timeless/restaurants/table_shapes/views.py

```diff
@@ -1,45 +1,53 @@
-"""Floors views module.
-@todo #72:30min Continue implementing list(), create(), edit() and
- delete() methods, using SQLAlchemy and TableShape model. In the index page it
- should be possible to sort and filter for every column. Table shape management
- page should be accessed by the Location page. Update html templates when
- methods are implemented. Create more tests for all methods.
+"""TableShape views module.
+ @todo #162:30min Update html templates when all
+ methods are implemented.
 """
 from flask import (
     Blueprint, flash, redirect, render_template, request, url_for
 )
 
+from timeless.db import DB
+from timeless.restaurants import models
+from timeless.restaurants.table_shapes import forms
+
+
 bp = Blueprint("table_shape", __name__, url_prefix="/table_shapes")
 
 
 @bp.route("/")
 def list():
-    """ List all table shapes"""
-    table_shapes = [{
-        "id": 1,
-        "description": "This is a test",
-        "picture": "shape.png"
-        }]
+    """List all table shapes
+    @todo #162:30min Implement ordering and sorting of TableShape model for
+    all columns. Do not forget write tests.
+    """
     return render_template(
-        "restaurants/table_shapes/list.html", table_shapes=table_shapes
-    )
+        "restaurants/table_shapes/list.html",
+        table_shapes=models.TableShape.query.all())
 
 
 @bp.route("/create", methods=("GET", "POST"))
 def create():
     """ Create new table shape"""
-    if request.method == "POST":
-        flash("Create not yet implemented")
-    action = "create"
+    form = forms.TableShapeForm(request.form)
+
+    if request.method == 'POST' and form.validate():
```

Comment:
@vryazanov I see form.validate() is called here, but where are the validation rules actually set?

Reply:
in `TableShapeForm` forms. That form is looking at `TableShape` model (see `class Meta` inside the form). So the form validates data according to the SQL-Alchemy model